### PR TITLE
IDEX-2886: added method to read specific file content to machine API

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java [hunk 1]
```diff
@@ -72,7 +76,7 @@
     @Inject
     public MachineService(MachineManager machineManager) {
         this.machineManager = machineManager;
-        this.dtoFactory = DtoFactory.getInstance();
+        this.dtoFactory = getInstance();
```
Comment (garagatyi) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineService.java]: remove this string

**Comment by garagatyi** [comment]
PR contains too much changes that are not related to the topic. It's hard to analyze PR because of that.
@akorneta please next time make such changes in source branch and merge to your branch to help reviewers reduce time needed to make review. I know that we often do the same, but we all should try to avoid that.

**Comment by skabashnyuk** [comment]
Javadoc plz

File: platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java [hunk 2]
```diff
@@ -701,6 +701,81 @@ public void run() {
...
+
+    /**
+     * Gets machines states.
+     *
+     * @return list of machine states
+     * @throws MachineException
+     *         if any error occur with machine registry
+     */
+    public List<MachineImpl> getMachinesStates() throws MachineException {
+        return machineRegistry.getStates();
```
Comment (skabashnyuk) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: Is this a copy or reference?

Comment (akorneta) [inline platform-api/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java]: references

**Pull request merged**